src/managers/KeybindManager.cpp

```diff
@@ -2013,6 +2014,162 @@ void CKeybindManager::pass(std::string regexp) {
+    const auto ARGS = CVarList(args, 3);
+    if (ARGS.size() != 3) {
+        Debug::log(ERR, "sendshortcut: invalid args");
+        return;
+    }
+
+    const auto MOD = g_pKeybindManager->stringToModMask(ARGS[0]);
+    const auto KEY = ARGS[1];
+    uint32_t   keycode;
+    bool       isMouse = 0;
+
+    //oriented from parseKey in ConfigManager
+    if (isNumber(KEY) && std::stoi(KEY) > 9) {
+        keycode = std::stoi(KEY);
+    } else if (KEY.compare(0, 5, "code:") == 0 && isNumber(KEY.substr(5))) {
+        keycode = std::stoi(KEY.substr(5));
+    } else if (KEY.compare(0, 6, "mouse:") == 0 && isNumber(KEY.substr(6))) {
+        keycode = std::stoi(KEY.substr(6));
+        isMouse = 1;
+        if (keycode < 272) {
+            Debug::log(ERR, "sendshortcut: invalid mouse button");
+            return;
+        }
+    } else {
+
+        // here, we need to find the keycode from the key name
+        // this is not possible through xkb's lib, so we need to iterate through all keycodes
+        // once found, we save it to the cache
+
+        const auto KEYSYM = xkb_keysym_from_name(KEY.c_str(), XKB_KEYSYM_CASE_INSENSITIVE);
+        keycode           = 0;
+
+        const auto KB            = g_pSeatManager->keyboard;
+        const auto KEYPAIRSTRING = std::to_string(KB) + KEY;
+
+        // check if KEY is in m_mKeyToCodeCache
+        if (g_pKeybindManager->m_mKeyToCodeCache.find(KEYPAIRSTRING) == g_pKeybindManager->m_mKeyToCodeCache.end()) {
+            // find keycode and add to cache
+            // get current window
+            xkb_keymap*   km = KB->wlr()->keymap;
+            xkb_state*    ks = KB->xkbTranslationState;
+
+            xkb_keycode_t keycode_min, keycode_max;
+            keycode_min = xkb_keymap_min_keycode(km);
+            keycode_max = xkb_keymap_max_keycode(km);
+
+            for (xkb_keycode_t kc = keycode_min; kc <= keycode_max; ++kc) {
+                // Iterate through all the layers and levels of the key to find matching keysym
+
+                xkb_keysym_t sym = xkb_state_key_get_one_sym(ks, kc);
+
+                if (sym == KEYSYM) {
+                    keycode = kc; // found the matching keycode
+                    //save it to cache
+                    g_pKeybindManager->m_mKeyToCodeCache[KEYPAIRSTRING] = keycode;
+                }
+            }
+
+            if (keycode == 0) { //wasnt found
+                Debug::log(ERR, "sendshortcut: key not found");
+                return;
+            }
+
+        } else {
+            keycode = g_pKeybindManager->m_mKeyToCodeCache[KEYPAIRSTRING];
+        }
+    }
+
+    if (keycode == 0) {
+        Debug::log(ERR, "sendshortcut: invalid key");
+        return;
+    }
+
+    const std::string regexp = ARGS[2];
+
+    PHLWINDOW         PWINDOW          = nullptr;
+    const auto        LASTWINDOWEXISTS = g_pCompositor->m_pLastWindow.lock();
+
+    if (!LASTWINDOWEXISTS) {
+        Debug::log(ERR, "sendshortcut: no window to return to");
```

Comment:
this is not fatal, is it? Just unfocus afterwards in that case

Reply:
Is unfocus just putting the focus to the last focused window? If so, I would do that

Reply:
unfocus is putting focus on nothing

Reply:
What should I do then exactly in this case? Unfocus before focusing the old window?

Reply:
if there is no old window to return to, after sending the shortcut, unfocus?

Reply:
this is before changing the focus

Reply:
it will not perform a sendshortcut if there is no focused window

Reply:
ahhh i get it :)

Reply:
Done